crates/sui-adapter/src/execution_mode.rs

```diff
@@ -46,7 +46,7 @@ pub trait ExecutionMode {
         result: Self::ExecutionResult,
     );
 }
-
+#[derive(Clone)]
```

Comment:
nit, newline here between the `}` and the `#[derive` 
And you should also give it `Copy` if you are going to give it Clone

Reply:
And I don't see any cases where you are calling clone, do you need this?

Reply:
don't need this anymore with PhantomData

Reply:
With phantom data, you shouldn't need the clone right?

Reply:
Spent too much time in Move it seems :) This is how phantom works for Move

Reply:
Don't much mind it having clone, just find it surprising. Could you also give it `Copy` too? no reason for it not to have it